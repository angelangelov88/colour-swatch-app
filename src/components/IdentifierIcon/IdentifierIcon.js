import React from 'react'
import classNames from 'classnames'
import PropTypes from 'prop-types'

const IdentifierIcon = (props) => {
  let {
    color,
    className,
    size,
    type,
    Icon,
    statusIndicator,
    status,
    initials
  } = props

  const classes = classNames(
    "flex items-center justify-center relative rounded-full font-ibm text-white uppercase",
    size === "xs" && "w-8 h-8 text-xs",
    size === "sm" && "w-9 h-9 text-base",
    size === "md" && "w-10 h-10 text-lg",
    size === "lg" && "w-11 h-11 text-xl",
    color === "gray" && "bg-gray-60",
    color === "primary" && "bg-primary-50",
    color === "purple" && "bg-purple-50",
    color === "blue" && "bg-blue-50",
    color === "lightblue" && "bg-lightblue-60",
    color === "cyan" && "bg-cyan-60",
    color === "teal" && "bg-teal-60",
    color === "green" && "bg-green-60",
    color === "yellow" && "bg-yellow-60",
    color === "red" && "bg-red-50",
    className
  )

  const dotClasses = classNames(
    "absolute bottom-0 right-0 rounded-full shadow-[0_0_0_2px_rgb(255,255,255)]",
    (size === "xs" || size === "sm") ? "w-2 h-2" : "w-2.5 h-2.5",
    !statusIndicator && "hidden",
    status ? "bg-green-50" : "bg-gray-30"
  )

  return (
    <div className={classes}>
      { type === "initials" ?
        <span>{initials}</span>:
        <Icon className="w-full p-2" />
      }
      <div className={dotClasses}></div>
    </div>
  )
}

IdentifierIcon.defaultProps = {
  color: "primary",
  size: "lg",
  type: "initials",
  statusIndicator: true,
  status: true,
  initials: "kl"
}

IdentifierIcon.propTypes = {
  className: PropTypes.string,
  color: PropTypes.oneOf([
    "gray",
    "primary",
    "purple",
    "blue",
    "lightblue",
    "cyan",
    "teal",
    "green",
    "yellow",
    "red",
    "void",
  ]),
  size: PropTypes.oneOf(["xs", "sm", "md", "lg"]),
  type: PropTypes.oneOf(["initials", "icon"]),
  Icon: PropTypes.object,
  statusIndicator: PropTypes.bool,
  status: PropTypes.bool,
  initials: PropTypes.string
}

export { IdentifierIcon }
