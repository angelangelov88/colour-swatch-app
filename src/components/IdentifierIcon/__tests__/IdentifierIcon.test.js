import renderer from 'react-test-renderer'
import {IdentifierIcon} from "../IdentifierIcon"
import { DocumentIcon } from '@heroicons/react/outline'

it('should render with default props', () => {
  const component = renderer.create(
    <IdentifierIcon />,
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
});


it('should render with color gray', () => {
  const component = renderer.create(
    <IdentifierIcon 
      color="gray" 
    />,
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
});


it('should render with color primary', () => {
  const component = renderer.create(
    <IdentifierIcon 
      color="primary" 
    />,
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
});


it('should render with color purple', () => {
  const component = renderer.create(
    <IdentifierIcon 
      color="purple" 
    />,
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
});


it('should render with color blue', () => {
  const component = renderer.create(
    <IdentifierIcon 
      color="blue" 
    />,
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
});


it('should render with color lightblue', () => {
  const component = renderer.create(
    <IdentifierIcon 
      color="lightblue" 
    />,
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
});


it('should render with color cyan', () => {
  const component = renderer.create(
    <IdentifierIcon 
      color="cyan" 
    />,
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
});


it('should render with color teal', () => {
  const component = renderer.create(
    <IdentifierIcon 
      color="teal" 
    />,
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
});


it('should render with color green', () => {
  const component = renderer.create(
    <IdentifierIcon 
      color="green" 
    />,
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
});


it('should render with color yellow', () => {
  const component = renderer.create(
    <IdentifierIcon 
      color="yellow" 
    />,
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
});


it('should render with color red', () => {
  const component = renderer.create(
    <IdentifierIcon 
      color="red" 
    />,
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
});


it('should render with size xs', () => {
  const component = renderer.create(
    <IdentifierIcon 
      size="xs"
    />,
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('should render with size sm', () => {
  const component = renderer.create(
    <IdentifierIcon 
      size="sm"
    />,
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})


it('should render with size md', () => {
  const component = renderer.create(
    <IdentifierIcon 
      size="md"
    />,
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})


it('should render with size lg', () => {
  const component = renderer.create(
    <IdentifierIcon 
      size="lg"
    />,
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})


it('should render with type icon and DocumentIcon', () => {
  const component = renderer.create(
    <IdentifierIcon 
      type="icon"
      Icon={DocumentIcon}
    />,
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})


it('should render with type initials', () => {
  const component = renderer.create(
    <IdentifierIcon 
      type="initials"
    />,
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})


it('should render with status indicator', () => {
  const component = renderer.create(
    <IdentifierIcon 
      statusIndicator={true}
    />,
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})


it('should render without status indicator', () => {
  const component = renderer.create(
    <IdentifierIcon 
      statusIndicator={false}
    />,
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})


it('should render with status true', () => {
  const component = renderer.create(
    <IdentifierIcon 
      status={true}
    />,
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})


it('should render with status false', () => {
  const component = renderer.create(
    <IdentifierIcon 
      status={false}
    />,
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})


it('should render with initials DA', () => {
  const component = renderer.create(
    <IdentifierIcon 
      initials="DA"
    />,
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})


it('should render with initials PH', () => {
  const component = renderer.create(
    <IdentifierIcon 
      initials="PH"
    />,
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})



