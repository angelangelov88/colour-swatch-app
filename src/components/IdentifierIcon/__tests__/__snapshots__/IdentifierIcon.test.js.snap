// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with default values - primary color, lg, initials, status online but have initials rt instead of the default 1`] = `
<div
  className="flex items-center justify-center relative rounded-full font-ibm text-white uppercase w-11 h-11 text-xl bg-primary-50"
>
  <span>
    Rt
  </span>
  <div
    className="absolute bottom-0 right-0 rounded-full  outline-2 outline outline-white w-2.5 h-2.5 bg-green-50"
  />
</div>
`;

exports[`should render with initals RR, default color, small size, and status offline/gray/false 1`] = `
<div
  className="flex items-center justify-center relative rounded-full font-ibm text-white uppercase w-9 h-9 text-base bg-primary-50"
>
  <span>
    RR
  </span>
  <div
    className="absolute bottom-0 right-0 rounded-full  outline-2 outline outline-white w-2 h-2 bg-gray-30"
  />
</div>
`;

exports[`should render with props color="blue" size="sm" type="icon" Icon={DocumentIcon} statusIndicator={true} status={true} 1`] = `
<div
  className="flex items-center justify-center relative rounded-full font-ibm text-white uppercase w-9 h-9 text-base bg-blue-50"
>
  <svg
    aria-hidden="true"
    className="w-full p-2"
    fill="none"
    stroke="currentColor"
    strokeWidth={2}
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M7 21h10a2 2 0 002-2V9.414a1 1 0 00-.293-.707l-5.414-5.414A1 1 0 0012.586 3H7a2 2 0 00-2 2v14a2 2 0 002 2z"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
  <div
    className="absolute bottom-0 right-0 rounded-full  outline-2 outline outline-white w-2 h-2 bg-green-50"
  />
</div>
`;

exports[`should render with props color="purple" size="lg" type="initials" statusIndicator={true} status={true} initials="gJ" 1`] = `
<div
  className="flex items-center justify-center relative rounded-full font-ibm text-white uppercase w-11 h-11 text-xl bg-purple-50"
>
  <span>
    gJ
  </span>
  <div
    className="absolute bottom-0 right-0 rounded-full  outline-2 outline outline-white w-2.5 h-2.5 bg-green-50"
  />
</div>
`;

exports[`should render with props color=red and Document icon 1`] = `
<div
  className="flex items-center justify-center relative rounded-full font-ibm text-white uppercase w-11 h-11 text-xl bg-red-50"
>
  <svg
    aria-hidden="true"
    className="w-full p-2"
    fill="none"
    stroke="currentColor"
    strokeWidth={2}
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M7 21h10a2 2 0 002-2V9.414a1 1 0 00-.293-.707l-5.414-5.414A1 1 0 0012.586 3H7a2 2 0 00-2 2v14a2 2 0 002 2z"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
  <div
    className="absolute bottom-0 right-0 rounded-full  outline-2 outline outline-white w-2.5 h-2.5 bg-green-50"
  />
</div>
`;

exports[`should render with props type="icon" Icon={DocumentIcon} no status 1`] = `
<div
  className="flex items-center justify-center relative rounded-full font-ibm text-white uppercase w-11 h-11 text-xl bg-primary-50"
>
  <svg
    aria-hidden="true"
    className="w-full p-2"
    fill="none"
    stroke="currentColor"
    strokeWidth={2}
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M7 21h10a2 2 0 002-2V9.414a1 1 0 00-.293-.707l-5.414-5.414A1 1 0 0012.586 3H7a2 2 0 00-2 2v14a2 2 0 002 2z"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
  <div
    className="absolute bottom-0 right-0 rounded-full  outline-2 outline outline-white w-2.5 h-2.5 hidden bg-green-50"
  />
</div>
`;

exports[`should render with props type="icon" Icon={DocumentIcon} statusIndicator={true} status={false} 1`] = `
<div
  className="flex items-center justify-center relative rounded-full font-ibm text-white uppercase w-11 h-11 text-xl bg-primary-50"
>
  <svg
    aria-hidden="true"
    className="w-full p-2"
    fill="none"
    stroke="currentColor"
    strokeWidth={2}
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M7 21h10a2 2 0 002-2V9.414a1 1 0 00-.293-.707l-5.414-5.414A1 1 0 0012.586 3H7a2 2 0 00-2 2v14a2 2 0 002 2z"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
  <div
    className="absolute bottom-0 right-0 rounded-full  outline-2 outline outline-white w-2.5 h-2.5 bg-gray-30"
  />
</div>
`;

exports[`should render with props type="initials" statusIndicator={false} initials="AH" 1`] = `
<div
  className="flex items-center justify-center relative rounded-full font-ibm text-white uppercase w-11 h-11 text-xl bg-primary-50"
>
  <span>
    AH
  </span>
  <div
    className="absolute bottom-0 right-0 rounded-full  outline-2 outline outline-white w-2.5 h-2.5 hidden bg-green-50"
  />
</div>
`;

exports[`should render with red color, document icon and no status shown 1`] = `
<div
  className="flex items-center justify-center relative rounded-full font-ibm text-white uppercase w-11 h-11 text-xl bg-red-50"
>
  <svg
    aria-hidden="true"
    className="w-full p-2"
    fill="none"
    stroke="currentColor"
    strokeWidth={2}
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M7 21h10a2 2 0 002-2V9.414a1 1 0 00-.293-.707l-5.414-5.414A1 1 0 0012.586 3H7a2 2 0 00-2 2v14a2 2 0 002 2z"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
  <div
    className="absolute bottom-0 right-0 rounded-full  outline-2 outline outline-white w-2.5 h-2.5 hidden bg-gray-30"
  />
</div>
`;

exports[`should render with small size, initials AF, status shown but offline and purple color 1`] = `
<div
  className="flex items-center justify-center relative rounded-full font-ibm text-white uppercase w-9 h-9 text-base bg-purple-50"
>
  <span>
    af
  </span>
  <div
    className="absolute bottom-0 right-0 rounded-full  outline-2 outline outline-white w-2 h-2 bg-gray-30"
  />
</div>
`;

exports[`should render with yellow color, document icon and online status displayed 1`] = `
<div
  className="flex items-center justify-center relative rounded-full font-ibm text-white uppercase w-11 h-11 text-xl bg-yellow-60"
>
  <svg
    aria-hidden="true"
    className="w-full p-2"
    fill="none"
    stroke="currentColor"
    strokeWidth={2}
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M7 21h10a2 2 0 002-2V9.414a1 1 0 00-.293-.707l-5.414-5.414A1 1 0 0012.586 3H7a2 2 0 00-2 2v14a2 2 0 002 2z"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
  <div
    className="absolute bottom-0 right-0 rounded-full  outline-2 outline outline-white w-2.5 h-2.5 bg-green-50"
  />
</div>
`;
